### PR TITLE
Replace travis with sbt-github-actions

### PR DESCRIPTION
**Pull request by mdedetrich**
Self explanatory. Actual publishing will be integrated later

### CODE REVIEW

**Comment by mdedetrich** [comment]
Uh what the, this seems to be failing because its missing a `/` in the local resolver, see

```
[error]   not found: /home/runner/.ivy2/localcom.github.sbt/paradox-material-theme/0.6.0-SNAPSHOT/ivys/ivy.xml
```

from https://github.com/sbt/sbt-paradox-material-theme/actions/runs/7825274770/job/21349193981?pr=48#step:10:45

**Comment by mdedetrich** [comment]
@eed3si9n Sorry to ping but do you have any possible clue what could be going on here, I have never experienced this before?

If you run `sbt '++ 2.12.18' test scripted makeSite` locally you get no issues however in CI the sbt resolver seems to be missing the `/` which means it cannot resolve the `publishLocal`'ed plugin?

File: publish.sbt [hunk 1]
```diff
@@ -41,6 +41,28 @@ inThisBuild(
     dynver := {
       val d = new java.util.Date
       sbtdynver.DynVer.getGitDescribeOutput(d).mkVersion(versionFmt, fallbackVersion(d))
-    }
+    },
+    githubWorkflowBuild := Seq(WorkflowStep.Sbt(List("test", "scripted", "makeSite"))),
```
Comment (eed3si9n) [inline publish.sbt:45]:
```suggestion
    githubWorkflowBuild := Seq(WorkflowStep.Sbt(List("test", "sbt-paradox-material-theme/scripted", "makeSite"))),
```

**Review by eed3si9n — COMMENTED** [review]
Not sure if this would cause the `local` without slash, but this might be worth trying.

**Comment by mdedetrich** [comment]
> Not sure if this would cause the `local` without slash, but this might be worth trying.

Nope same issue

```
[error] sbt.librarymanagement.ResolveException: Error downloading com.github.sbt:paradox-material-theme:0.6.0-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: /home/runner/.ivy2/localcom.github.sbt/paradox-material-theme/0.6.0-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/com/github/sbt/paradox-material-theme/0.6.0-SNAPSHOT/paradox-material-theme-0.6.0-SNAPSHOT.pom
```

**Comment by mkurz** [comment]
Hmm.. I think I have seen this `/` thing before... Need to think harder what the problem was back then... I will try to investigate when I find time...
@mdedetrich pushed a commit to upgrade some dependencies, hope that is ok (I think the scripted tests would fail without it anyway).

**Comment by mkurz** [comment]
Maybe an environment variable causing this?

**Comment by mdedetrich** [comment]
> Hmm.. I think I have seen this `/` thing before... Need to think harder what the problem was back then... I will try to investigate when I find time...
> @mdedetrich pushed a commit to upgrade some dependencies, hope that is ok (I think the scripted tests would fail without it anyway).

Can we update the dependencies in another PR? Pekko cannot use this otherwise because of the JDK 8 issue I talked about elsewhere.

**Comment by mdedetrich** [comment]
> @mdedetrich pushed a commit to upgrade some dependencies, hope that is ok (I think the scripted tests would fail without it anyway).

I dropped this commit with the reasoning being described in https://github.com/lightbend/paradox/issues/601. It would actually be ideal to release sbt-paradox-material-theme against sbt-paradox/sbt-paradox theme 0.9.x that is published in maven central and then do a future release with these dependencies updated to the latest.


Also the scripted tests pass as is fine locally, its literally only CI where we are getting this problem.

**Comment by mdedetrich** [comment]
So @eed3si9n has managed to find out the root cause of why `compile`/`update` was failing in CI, see https://github.com/sbt/sbt-paradox-material-theme/pull/53. I have updated this PR with the fix and everything is working as expected.

@mkurz

**Review by eed3si9n — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -11,6 +12,11 @@ lazy val root = project("paradox-material-theme-parent", file("."))
     Compile / paradoxNavigationDepth := 3,
     makeSite := makeSite.dependsOn(Compile / paradox).value,
     paradoxMaterialTheme / version := version.value,
+    // this is to avoid triggering update, which will fail due to be the build using an
+    // intertwined dependency pattern, see
+    // https://stackoverflow.com/questions/37424513/intertwined-dependencies-between-sbt-plugin-and-projects-within-multi-project-bu
+    Compile / compile := empty,
+    Compile / test := empty,
```
Comment (eed3si9n) [inline build.sbt:19]:
```suggestion
    Compile / test := (),
```

**Pull request merged**